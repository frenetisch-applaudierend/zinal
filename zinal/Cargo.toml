[package]
name = "zinal"
version = "0.2.1"
authors = ["Markus Gasser <markus.gasser@mailbox.org>"]
edition = "2021"
description = "HTML templating library for Rust programs"
repository = "https://github.com/frenetisch-applaudierend/zinal"
license = "MIT"
keywords = ["html", "template", "components"]
categories = ["template-engine", "web-programming"]
workspace = ".."

[package.metadata.docs.rs]
all-features = true

[dependencies]
zinal_derive = { path = "../zinal_derive", version = "0.2.1", optional = true }

[dev-dependencies]
axum = "0.7"
trybuild = "1.0"

[features]
default = ["derive"]

# Enable deriving the Template trait
derive = ["dep:zinal_derive"]

# Generate axum IntoResponse impls for derived templates
axum = ["zinal_derive?/axum"]
