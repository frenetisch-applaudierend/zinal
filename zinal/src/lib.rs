#![deny(unsafe_code)]

mod children;
mod context;
mod escaper;
mod renderable;
mod template;

pub mod builder;

pub use children::*;
pub use context::*;
pub use escaper::*;
pub use renderable::*;
pub use template::*;

#[cfg(feature = "derive")]
#[cfg_attr(docsrs, doc(cfg(feature = "derive")))]
pub use zinal_derive::Template;

#[cfg(test)]
mod tests;
