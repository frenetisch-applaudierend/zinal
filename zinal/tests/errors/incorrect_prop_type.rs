use zinal::*;

#[derive(Template)]
#[template("<Inner prop={{self.prop}} />")]
struct Outer {
    prop: u8,
}

#[derive(Template)]
#[template("{{self.prop}}")]
struct Inner {
    prop: String,
}

fn main() {
    println!(
        "{}",
        Example { prop: 8 }
            .render_to_string()
            .expect("Should not fail")
    );
}
