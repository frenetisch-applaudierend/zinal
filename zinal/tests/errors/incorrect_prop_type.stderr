error[E0422]: cannot find struct, variant or union type `Example` in this scope
  --> tests/errors/incorrect_prop_type.rs:18:9
   |
18 |         Example { prop: 8 }
   |         ^^^^^^^ not found in this scope

error[E0308]: mismatched types
  --> tests/errors/incorrect_prop_type.rs:3:10
   |
3  | #[derive(Template)]
   |          ^^^^^^^^- help: try using a conversion method: `.to_string()`
   |          |
   |          expected `String`, found `u8`
   |          arguments to this method are incorrect
   |
note: method defined here
  --> tests/errors/incorrect_prop_type.rs:12:5
   |
9  | #[derive(Template)]
   |          --------
...
12 |     prop: String,
   |     ^^^^
   = note: this error originates in the derive macro `Template` (in Nightly builds, run with -Z macro-backtrace for more info)
